### ISSUE TITLE
[Bug]: Today's FreeDesktop SDK 24.08 update causes SSL connections to fail

### ISSUE THREAD

**Issue by NoTuxNoBux** [flathub/com.unity.UnityHub#127]
### What Happened

Today (2024-10-10) an update to the FreeDesktop SDK 24.08 was pushed, and this update breaks SSL connections with `HTTPClient` amongs other things inside the Unity Editor.

The exact error appears to be `CryptographicException: Chain instance is empty.`.

### How to Reproduce

1. Open a Unity project with Unity 6000.0.11f1 (other versions such as 6000.0.22f1 also experience this).
2. Observe package manager error.

### Operating System

Arch Linux x86-64

### Anything else?

After opening the Unity editor the package manager will try to initialize and fail:

```
[Package Manager Window] Error while getting auth code: System.InvalidOperationException: Failed to call Unity ID to get auth code.
UnityEditor.AsyncHTTPClient:Done (UnityEditor.AsyncHTTPClient/State,int) (at /home/bokken/build/output/unity/unity/Editor/Mono/AsyncHTTPClient.cs:248)
```

We also use ZeroQL, which uses plain .NET `HTTPClient` under the hood, and it also started failing to establish SSL connections. The error is somewhat more verbose here:

```
CryptographicException: Chain instance is empty.
Mono.Unity.UnityTlsContext.ProcessHandshake () (at <fe4d66692fdb4d8e879dd148913ac75a>:0)
Mono.Net.Security.MobileAuthenticatedStream.ProcessHandshake (Mono.Net.Security.AsyncOperationStatus status, System.Boolean renegotiate) (at <fe4d66692fdb4d8e879dd148913ac75a>:0)
(wrapper remoting-invoke-with-check) Mono.Net.Security.MobileAuthenticatedStream.ProcessHandshake(Mono.Net.Security.AsyncOperationStatus,bool)
Mono.Net.Security.AsyncHandshakeRequest.Run (Mono.Net.Security.AsyncOperationStatus status) (at <fe4d66692fdb4d8e879dd148913ac75a>:0)
Mono.Net.Security.AsyncProtocolRequest.ProcessOperation (System.Threading.CancellationToken cancellationToken) (at <fe4d66692fdb4d8e879dd148913ac75a>:0)
Rethrow as AuthenticationException: Authentication failed, see inner exception.
Mono.Net.Security.MobileAuthenticatedStream.ProcessAuthentication (System.Boolean runSynchronously, Mono.Net.Security.MonoSslAuthenticationOptions options, System.Threading.CancellationToken cancellationToken) (at <fe4d66692fdb4d8e879dd148913ac75a>:0)
System.Net.Http.ConnectHelper.EstablishSslConnectionAsyncCore (System.IO.Stream stream, System.Net.Security.SslClientAuthenticationOptions sslOptions, System.Threading.CancellationToken cancellationToken) (at <0c94a492cead4dd9a87d20c16b237a7a>:0)
Rethrow as HttpRequestException: The SSL connection could not be established, see inner exception.
System.Net.Http.ConnectHelper.EstablishSslConnectionAsyncCore (System.IO.Stream stream, System.Net.Security.SslClientAuthenticationOptions sslOptions, System.Threading.CancellationToken cancellationToken) (at <0c94a492cead4dd9a87d20c16b237a7a>:0)
System.Threading.Tasks.ValueTask`1[TResult].get_Result () (at <467dcef8ceb84c1e92dc7c69d21858a8>:0)
System.Net.Http.HttpConnectionPool.CreateConnectionAsync (System.Net.Http.HttpRequestMessage request, System.Threading.CancellationToken cancellationToken) (at <0c94a492cead4dd9a87d20c16b237a7a>:0)
System.Threading.Tasks.ValueTask`1[TResult].get_Result () (at <467dcef8ceb84c1e92dc7c69d21858a8>:0)
System.Net.Http.HttpConnectionPool.WaitForCreatedConnectionAsync (System.Threading.Tasks.ValueTask`1[TResult] creationTask) (at <0c94a492cead4dd9a87d20c16b237a7a>:0)
System.Threading.Tasks.ValueTask`1[TResult].get_Result () (at <467dcef8ceb84c1e92dc7c69d21858a8>:0)
System.Net.Http.HttpConnectionPool.SendWithRetryAsync (System.Net.Http.HttpRequestMessage request, System.Boolean doRequestAuth, System.Threading.CancellationToken cancellationToken) (at <0c94a492cead4dd9a87d20c16b237a7a>:0)
System.Net.Http.RedirectHandler.SendAsync (System.Net.Http.HttpRequestMessage request, System.Threading.CancellationToken cancellationToken) (at <0c94a492cead4dd9a87d20c16b237a7a>:0)
System.Net.Http.HttpClient.FinishSendAsyncBuffered (System.Threading.Tasks.Task`1[TResult] sendTask, System.Net.Http.HttpRequestMessage request, System.Threading.CancellationTokenSource cts, System.Boolean disposeCts) (at <0c94a492cead4dd9a87d20c16b237a7a>:0)
...
```

I downgraded to the previous commit of the SDK for now, force-closed the Unity hub and editor and it is fixed again.

**Comment by wjt** [comment]
Are you interested in maintaining this Flatpak?

**Comment by NoTuxNoBux** [comment]
Turns out this is now also happening with the 23.08 SDK since yesterday :see_no_evil: . I also [reported this upstream](https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/issues/1781#note_2173034309).

**Comment by NoTuxNoBux** [comment]
Upstream fixed this for the 23.08 and 24.08 runtimes, see also the linked issue, so this can be closed. Thanks for following up!
